Feature/9 create button component

modous opened the pull request:
https://github.com/users/modous/projects/6/views/1?pane=issue&itemId=52761970

modous committed:
feat: added completed state

modous committed:
Fix: fixed the complete state

modous committed:
Fix: added right styling from figma

modous committed:
Feat: added React classnames + right styling

modous committed:
Fix: trashIcon only shows when hovering li

modous committed:
feat: added the edit input + saves on blur

modous committed:
Feat: added index.ts for shorter import + right file names + changed export default to just default

modous committed:
Feat: added trash icon from react library + added smooth hover animations

modous committed:
Feat: try to fix the edit text

modous committed:
feat: added exit editmode after pressing enter + styling

modous committed:
feat: added autoFocus + classnames to span

modous committed:
fix: styling on edit

modous committed:
fix: now u see the editTitle. instead of info in API

modous committed:
feat: added cancel edit when pressing escape

modous committed:
fix: added comment + changed variable name

modous committed:
feat: added button component

modous committed:
fix: removed :root font family style

modous committed:
feat: updated the Button fuction

modous committed:
fix: changed propname to onTitleChange

modous committed:
feat: added css reset file

modous committed:
feat: removed trashIcon component added it in TodoItem + used button instead of span

modous committed:
fix: styling on button

modous committed:
fix: styling for the on edit + removed trashbutton when onEdit + the title has bigger space to click on

modous committed:
feat: added classnames + ...props , now ready to be used as a component

modous committed:
fix: added componentprops

modous committed:
feat: added variant styling styles in button

modous committed:
feat: implemented the button component in TodoItem + styling

modous committed:
feat: added hover and focus states for button

modous committed:
fix: added outline so the icon does not move

montagamove commented on src/components/Checkbox/checkbox.tsx:
```diff
@@ -1,26 +1,31 @@
-  checked: boolean;
-  onChange: () => void;
+interface ICheckboxProps extends ComponentProps<"input"> {
+  onCheckedChange?: (isChecked: boolean) => void;
```
This `onCheckedChange` is not necessary because you're already in control of the `isChecked` property in the TodoItem component

montagamove commented on src/components/Checkbox/checkbox.tsx:
```diff
@@ -1,26 +1,31 @@
 
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    onChange?.(event);
+    onCheckedChange?.(event.target.checked);
```
You can remove this onCheckedChange

montagamove commented on src/components/button/Button.tsx:
```diff
@@ -0,0 +1,42 @@
+}: ButtonProps) {
+  return (
+    <button
+      className={classnames(
```
You can simplify this:

```
 <button
      type={type}
      className={classnames(
        styles.button,
        styles[variant],
        styles[size],
        className
      )}
      {...props}
    >
```

montagamove commented on src/components/button/Button.tsx:
```diff
@@ -0,0 +1,42 @@
+      type={type}
+      {...props}
+    >
+      {icon}
```
The icon is optional so you should render it conditionally: 

`{icon && icon} `

montagamove commented on src/components/button/Index.module.css:
```diff
@@ -0,0 +1,64 @@
+.primary {
+  width: 69px;
+  height: 40px;
+  padding: 12px, 16px, 12px, 16px;
```
`padding: 12px 16px;`

montagamove commented on src/components/button/Index.module.css:
```diff
@@ -0,0 +1,64 @@
+  box-shadow: 0px 1px 3px 0px #00000014;
+  transition: transform 0.3s ease;
+  display: none;
+}
```
Can you make sure that the styles that are common in both the primary and secondary button are put together. So like: 

```
.primary, 
.secondary {
--- common styles ---
} 

.primary {
--- specific styles ---
} 

.secondary {
--- specific styles --- 
}
```

montagamove commented on src/components/checkbox/Index.module.css:
```diff
@@ -0,0 +1,35 @@
+
+.checked:focus {
+  border: 2px solid #155dfb;
+}
```
You can do this:

```
.checkbox:focus,
.checked:focus {
  border: 2px solid #155dfb;
}
```

montagamove commented on src/components/todo-item/Index.module.css:
```diff
@@ -0,0 +1,75 @@
+.container {
+  padding: 8px, 16px, 8px, 16px;
```
`padding: 8px 16px;`

montagamove commented on src/components/todo-item/TodoItem.tsx:
```diff
@@ -0,0 +1,110 @@
+        })}
+        type="button"
+        variant="secondary"
+        size="md"
```
`md` is already your default so you don't have to specify it

modous committed:
fix: Implemented Montaga feedback on PR

modous committed:
fix: changed index.module.css name

modous committed:
fix: fixed some styling for the checkbox and delete button component

modous committed:
fix: added checkbox classnames

modous committed:
fix: removed dependencies i don't use + removed checkboxClass

I use css to define different styling voor de checkbox. I don't need the checkboxClas variable

modous committed:
fix: added types in todoitem

jaerpollux commented:
I just tried to use the button component and see the implementation is still very different from the design (check screenshot below).

Please try to follow the design exactly and make sure all different states, variants, sizes look like the design. The button should grow/shrink according to the children you pass. Let's discuss next steps on monday.

![CleanShot 2024-02-23 at 18 01 35](https://github.com/modous/Todo-challenge/assets/148754291/8d824d70-0737-4fd3-9d79-8e912a26ce40)


![image](https://github.com/modous/Todo-challenge/assets/148754291/fcb0cda1-2d5e-4ec3-8163-1ff88eb2fd2b)
